openedx/core/djangoapps/credit/api.py

```diff
@@ -0,0 +1,159 @@
+
+    Example:
+        >>> get_credit_requirements("course-v1-edX-DemoX-1T2015")
+                {
+                    requirements =
+                    [
+                        {
+                            "namespace": "verification",
+                            "name": "verification",
+                            "criteria": {},
+                        },
+                        {
+                            "namespace": "reverification",
+                            "name": "midterm",
+                            "criteria": {},
+                        },
+                        {
+                            "namespace": "proctored_exam",
+                            "name": "final",
+                            "criteria": {},
+                        },
+                        {
+                            "namespace": "grade",
+                            "name": "grade",
+                            "criteria": {"min_grade": 0.8},
+                        },
+                    ]
+                }
+
+    Returns:
+        Dict of requirements in the given namespace
+    """
+
+    requirements = CreditRequirement.get_course_requirements(course_key, namespace)
+    return [
+        {
+            "namespace": requirement.namespace,
+            "name": requirement.name,
+            "criteria": requirement.criteria
+        }
+        for requirement in requirements
+    ]
+
+
+def _get_requirements_to_disable(old_requirements, new_requirements):
+    """ Returns the ids of CreditRequirement to be disabled that are deleted from the courseware
+
+    Args:
+        old_requirements(QuerySet): QuerySet of CreditRequirement
+        new_requirements(list): List of requirements being added
+
+    Returns:
+        List of ids of CreditRequirement that are not in new_requirements
+    """
+    requirements_to_disable = []
+    for old_req in old_requirements:
+        found_flag = False
+        for req in new_requirements:
+            if req["namespace"] == old_req.namespace and req["name"] == old_req.name:
+                found_flag = True
+                break
+        if not found_flag:
+            requirements_to_disable.append(old_req.id)
+    return requirements_to_disable
+
+
+def _validate_requirements(requirements):
+    """ Validate the requirements
+
+    Args:
+        requirements(list): List of requirements
+
+    Returns:
+        List of strings of invalid requirements
+    """
+    invalid_requirements = []
+    for requirement in requirements:
+        invalid_params = []
+        if not requirement.get("namespace"):
+            invalid_params.append("namespace")
+        if not requirement.get("name"):
+            invalid_params.append("name")
+        if "criteria" not in requirement:
+            invalid_params.append("criteria")
+
+        if invalid_params:
+            invalid_requirements.append(str(requirement) + " has missing/invalid parameters: %s" % invalid_params)
```

Comment:
I think this might break for non-ASCII unicode.  This is safer:
```
invalid_requirements.append(
    u"{requirement} has missing/invalid parameters: {params}".format(
        requirement=requirement,
        parameters=invalid_params,
    )
)
```

(For string concatenation, please use `.format()` rather than `%s`.)